### PR TITLE
Miscellaneous arm64 vmm kernel changes

### PR DESCRIPTION
**Pull request by markjdb**
These address some of the issues I pointed out in the review, plus some additional bugs I found. If you're happy with the patches, please squash them into the main commit.

### CODE REVIEW

File: sys/arm64/vmm/vmm_arm64.c [hunk 1]
```diff
@@ -441,22 +441,23 @@ vmmops_modcleanup(void)
 
 	smp_rendezvous(NULL, arm_teardown_vectors, NULL, NULL);
 
-#ifdef INVARIANTS
```
Comment (zxombie) [inline sys/arm64/vmm/vmm_arm64.c:440]: Do we need to remove the stacks in the non-INVARIANTS case? It should only be needed to not hit an assert to check we've removed all entries. As we never call into the hypervisor after the above `smp_rendezvous` and are about to free all page tables removing the entries shouldn't be needed.

Comment (markjdb) [reply sys/arm64/vmm/vmm_arm64.c:440]: Hmm, I think you're right that this is not strictly necessary. But even so, why not just do the cleanup unconditionally? It seems quite strange to make this operation conditional on INVARIANTS, and it's fragile in the face of future changes to the vmmpmap_* interface.

Comment (zxombie) [reply sys/arm64/vmm/vmm_arm64.c:440]: It's only there to remove a false-positive in the memory leak `KASSERT` under `vmmpmap_fini`.

What future changes do you have planned that would make this needed?

Comment (markjdb) [reply sys/arm64/vmm/vmm_arm64.c:440]: I don't have any future changes planned, I just mean that the use of INVARIANTS here is making assumptions about the internals of vmmpmap_fini(). This is not how we generally use INVARIANTS elsewhere in the kernel, and this code is not performance-sensitive, so I still don't understand why we don't use the same code in INVARIANTS and non-INVARIANTS kernels.

If you insist I can drop this patch (and I do need to rework the commit message in any case), but this use of `#ifdef INVARIANTS` is wrong IMHO.

**Pull request merged**